include/mata/utils/ord-vector.hh

```diff
@@ -408,38 +424,45 @@ public:
         return result;
     }
 
-    static OrdVector Union(const OrdVector& lhs, const OrdVector& rhs) {
+    static void unite(const OrdVector& lhs, const OrdVector& rhs, OrdVector& result) {
         assert(lhs.vectorIsSorted());
         assert(rhs.vectorIsSorted());
 
-        OrdVector result;
-
-        auto lhs_it = lhs.begin();
-        auto rhs_it = rhs.vec_.begin();
-
-        while ((lhs_it != lhs.end()) || (rhs_it != rhs.end())) { // Until we get to the end of both vectors.
-            if (lhs_it == lhs.end()) { // If we are finished with the left-hand side vector.
-                result.push_back(*rhs_it);
-                ++rhs_it;
-            } else if (rhs_it == rhs.end()) { // If we are finished with the right-hand side vector.
-                result.push_back(*lhs_it);
-                ++lhs_it;
-            } else {
-                if (*lhs_it < *rhs_it) {
-                    result.push_back(*lhs_it);
-                    ++lhs_it;
-                } else if (*rhs_it < *lhs_it) {
-                    result.push_back(*rhs_it);
-                    ++rhs_it;
-                } else { // In case they are equal.
-                    result.push_back(*rhs_it);
-                    ++rhs_it;
-                    ++lhs_it;
-                }
-            }
-        }
+        result.reserve(lhs.size()+rhs.size());
+        std::set_union(lhs.vec_.begin(),lhs.vec_.end(),rhs.vec_.begin(),rhs.vec_.end(),std::back_inserter(result.vec_));
```

Comment:
I wonder if it would help if we check edge cases first? I.e.
```c++
if (lhs.empty()) {
  return rhs;
}
if (rhs.empty()) {
  return lhs;
}
```

Reply:
ok, makes sense